DifferenceGenerator/src/main/kotlin/algorithms/DivideAndConquerAligner.kt

```diff
@@ -138,18 +138,20 @@ class DivideAndConquerAligner<T>(private val algorithm: AlignmentAlgorithm<T>, p
      */
     private fun findMatches(): Array<Pair<Int, Int>> {
         val equals = ArrayList<Pair<Int, Int>>()
+        // optimize runtime and prevent swapped matches by keeping track of the minimum index
+        var minimum = -1
         // find all equal frames that are not marked as duplicates
         for (i in hashes1.indices) {
             // skip if marked as duplicate
             if (hashes1[i].isEmpty()) continue
-            for (j in hashes2.indices) {
+            for (j in minimum + 1 until hashes2.size) {
```

Comment:
This leads to the first match of a swapped pair being chosen. Was that the approach we decided to take or was it that we exclude both matches that are part of the swapping?

Reply:
we didn't really decide for an approach, the industry partner had no requirements here. But if we decide that we want to exclude both or thtow an error, then this can be changed. RN it is a bug so we need a fix